### ISSUE TITLE
Feeds: Show description for RSS feeds?

### ISSUE THREAD

**Issue by aaornrylow** [wet-boew/wet-boew#7618]
Hi,

Is it possible to include the RSS <description> field in the WET4 feeds widget? It was displayed by default in the WET3 version.

Thanks,
-Aaron

**Comment by crochefort** [comment]
are you talking about this http://wet-boew.github.io/v4.0-ci/demos/feeds/feeds-en.html ??

**Comment by aaornrylow** [comment]
Yes. That's the one.

**Comment by aaornrylow** [comment]
My first comment filtered it out because I used `<description>` without "code" markup, but just to be clear I was talking about the RSS description field.

**Comment by aaornrylow** [comment]
Any update on this issue?

**Comment by hsrudnicki** [comment]
Can be this closed or otherwise? Please update the status of this again, whether this remains still an issue? Thx

**Comment by aaornrylow** [comment]
This is still an issue.

**Comment by aaornrylow** [comment]
Any word on this? The WET3 version is currently broken ([even on your site](http://wet-boew.github.io/wet-boew-legacy/v3.1/demos/webfeeds/webfeeds-en.html)). It just shows the spinning "loading" symbol.
We'd like to move to the WET4 version but it's not very practical in our context without also being able to show the description field.

**Comment by kodecount** [comment]
[WET3 support was ended](http://wet-boew.github.io/wet-boew/docs/versions/rdmp-en.html):

> 2014-09-23	
> v4.0.6 released
> v3.1.15 released
> End of support for v3.1.x

Also WET 3 used ajax.googleapis.com which is now deprecated.

**Comment by kodecount** [comment]
On another note what is it your trying to get a feed for? Lot of companies slowly ending support for these type of feeds (like Facebook).

**Comment by aaornrylow** [comment]
We're not asking for further support on the WET3 version. The issue is with the current [WET**4** feeds plugin](http://wet-boew.github.io/v4.0-ci/demos/feeds/feeds-en.html) when using it to pull from a standard RSS (XML) feed ([here's the current feed on our website, for example](http://www.crtc.gc.ca/eng/rss/2017releases-jan.xml)). Unlike the WET3 version it does not have the option to display the **description** field from the RSS file. This field provides key info for us so we need it to display.

**Comment by LaurentGoderre** [comment]
I'm afraid that's not easily done in WET4

**Comment by duboisp** [comment]
@aaornrylow I understand your feature request but it might be only done for ATOM feed not RSS feed.

The ATOM feed equivalent of RSS description field is : ```summary```

Task:
* [ ] Modify the generic template in the feed widget to display ```summary``` if specified in the feed configuration
* [ ] Documents that new feature, add a working example.
* [ ] Assign someone for this request

@aaornrylow a side note. As per the [Standard on Web Interoperability](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=25875), Appendix B, your feed should be converted in the Atom Syndication Format ([RFC4287](https://tools.ietf.org/html/rfc4287))

**Comment by kodecount** [comment]
@aaornrylow If changing the feed is problematic you could try:

HTML:
```html
<section id="wb-feeds-custom" class="wb-feeds limit-5">
<h3>Releases Jan</h3>
<ul class="feeds-cont">
	<li>
		<a href="http://www.crtc.gc.ca/eng/rss/2017releases-jan.xml" rel="external">Releases Jan</a>
	</li>
</ul>
</section>
```

Javascript:
```
<script>
$( document ).on( "wb-feed-ready.wb-feeds", ".wb-feeds .feeds-cont", function( event ) {
	var url = "http://www.crtc.gc.ca/eng/rss/2017releases-jan.xml", limit, success = false;
	$.ajax({
		type: 'GET',
		url: 'https://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20feed%20where%20url%20%3D%20\'' + encodeURIComponent(decodeURIComponent(url)) + '\'%20limit%20' + (limit ? limit : 5) + '&format=json',
		dataType: 'json',
		success: function( data ) {
			success = true;
		}
	}).done(function( data ) {
		if (success) {
			var listItems = event.target.getElementsByTagName('li'),
				feedData = data.query.results.item;
			for ( var i = 0; i < listItems.length; i++ ) {
				var feedTitle = listItems[i].getElementsByTagName('a')[0].innerHTML,
					feedDataTitle = feedData[i].title,
					feedDataDescription = feedData[i].description,
					br = document.createElement("br"),
					span = document.createElement("span");
				if ( feedTitle.toUpperCase() == feedDataTitle.toUpperCase() ) {
					span.innerHTML = feedDataDescription;
					listItems[i].appendChild( br );
					listItems[i].appendChild( span );
				}
			}
		}
	});
});
</script>
```

**Comment by duboisp** [comment]
@kodecount I do not think that changing the feed is problematic. It's just that someone need to submit a pull request with the fix.

**Comment by aaornrylow** [comment]
Thanks for this. I found one bug so far. Maybe you can help me squash it.

If you change the feed limit to display all items (there are currently 19), there are 4 items in a row near the very end that don't display their descriptions:
![image](https://cloud.githubusercontent.com/assets/5074169/22220287/1248979a-e17c-11e6-9c75-20396dddb342.png)

This is because of the condition `if ( feedTitle.toUpperCase() == feedDataTitle.toUpperCase() )`.
For those 4 items, the `feedTitle` and `feedDateTitle` do not match.

Can you help me figure out why it would be different for only those 4 items?

**Comment by kodecount** [comment]
I've updated https://github.com/wet-boew/wet-boew/issues/7618#issuecomment-274126965 which should fix them bugs.

**Comment by duboisp** [comment]
Planned for the next release of WET (version 4.0.24), you will be able to achieve something similar by using the data-json plugin.

You will find a temporary working example here: http://universallabs.org/wet/labs/data-json/display-feed-issue7618/unmin/demos/data-json/feeds-layout-en.html

It will be simple as:

```html
<div data-wb-json='{
		"url": "https://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20feed%20where%20url%20%3D%20&#39;http%3A%2F%2Fwww.crtc.gc.ca%2Feng%2Frss%2F2017releases-jan.xml&#39;%20limit%2019&amp;format=json#/query/results/item",
		"mapping": [
			{ "selector": "h2", "value": "/title", "placeholder": "" },
			{ "selector": "a", "value": "/description" },
			{ "selector": "a", "value": "/link", "attr": "href" },
			{ "selector": "span", "value": "/pubDate", "placeholder": "[[pubdate]]" }
		]
	}'>

	<template>
		<h2></h2>
		<p><a href=""></a> <span>Publish on [[pubdate]]</span></p>
	</template>
</div>
```
then it render like:

![feeds-data-json](https://cloud.githubusercontent.com/assets/1676030/22234880/88988fc0-e1c9-11e6-86a7-1549cb0980eb.png)

**Comment by LaurentGoderre** [comment]
That data-json plugin sounds redundant to me. If you want binding like this, shouldn't you use something like angular instead of creating a pseudo angular?

**Comment by duboisp** [comment]
@LaurentGoderre for better recording purpose let have that discussion when I will submit the official PR. It should be submitted by next week.

**Comment by duboisp** [comment]
@aaornrylow Once the Pull Request #7875 are merged-in, you will have more flexibility and more control on how you can layout your feed. See the Template HTML 5 working example in the pull request description.

**Comment by duboisp** [comment]
@aaornrylow if you are using GCWeb theme, you can use the data-json plugin to do it. See the "Layout a RSS feeds" working example here: http://wet-boew.github.io/themes-dist/GCWeb/demos/data-json/template-en.html
